example/app/api.go

```diff
@@ -0,0 +1,119 @@
+package app
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"reflect"
+
+	"github.com/DE-labtory/cleisthenes/core"
+
+	kitendpoint "github.com/go-kit/kit/endpoint"
+	kitlog "github.com/go-kit/kit/log"
+	kithttp "github.com/go-kit/kit/transport/http"
+	"github.com/gorilla/mux"
+)
+
+type ErrIllegalArgument struct {
+	Reason string
+}
+
+func (e ErrIllegalArgument) Error() string {
+	return fmt.Sprintf("err illegal argument: %s", e.Reason)
+}
+
+type endpoint struct {
+	logger kitlog.Logger
+	hbbft  core.Hbbft
+}
+
+func newEndpoint(hbbft core.Hbbft, logger kitlog.Logger) *endpoint {
+	return &endpoint{
+		logger: logger,
+		hbbft:  hbbft,
+	}
+}
+
+func NewApiHandler(hbbft core.Hbbft, logger kitlog.Logger) http.Handler {
+	endpoint := newEndpoint(hbbft, logger)
+	r := mux.NewRouter()
+
+	opts := []kithttp.ServerOption{
+		kithttp.ServerErrorLogger(logger),
+	}
+
+	r.Methods("GET").Path("/healthz").HandlerFunc(func(w http.ResponseWriter, request *http.Request) {
+		logger.Log("method", "GET", "endpoint", "healthz")
+		w.Write([]byte("up"))
+	})
+
+	r.Methods("POST").Path("/tx").Handler(kithttp.NewServer(
+		endpoint.proposeTx,
+		decodeProposeTxRequest,
+		encodeResponse,
+		opts...,
+	))
+	return r
+}
+
+func (e *endpoint) proposeTx(ctx context.Context, request interface{}) (interface{}, error) {
+	e.logger.Log("endpoint", "proposeTx")
+
+	f := e.makeProposeTxEndpoint()
+	response, err := f(ctx, request)
+	if err != nil {
+		e.logger.Log("endpoint", "proposeTx", "err", err.Error())
+	}
+	return response, err
+}
+
+func (e *endpoint) makeProposeTxEndpoint() kitendpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		req := request.(ProposeTxRequest)
+		return ProposeTxResponse{}, e.hbbft.Propose(req.Transaction)
```

Comment:
Maybe we need validation logic for validate the transaction. 
Considering the future, anyway we need validation logic.

Reply:
`TxQueueManager` charge of that responsibility